File: Cine/gestorAplicación/cine/Usuario.java
package gestorAplicación.cine;

import java.util.ArrayList;

//import baseDatos.Deserializador;

import java.io.*;

import java.util.HashMap; 

public class Usuario implements Serializable {
	//atributos
	private static final long serialVersionUID = 1L;
	
	private String Nombre;
	private int Cedula;
	private boolean Membresia;
	private int Saldo;
	private ArrayList<Boleta> Boletas = new ArrayList<Boleta>();
	private HashMap<String,Integer> Carrito = new HashMap<String,Integer>();
	private static int cantidadUsuarios;
	
	public static int getCantidadUsuarios() {
		return cantidadUsuarios;
		
	}
	
	public Usuario() {
		
		this("Sin nombre", 0, 0);
		
	//	Deserializador.deserializar(this);	
	
	}
//constructores		
	public Usuario (String Nombre, int Cedula, int Saldo) {
		this.Nombre = Nombre;
		this.Cedula = Cedula;
		this.Membresia = false;
		this.Saldo = Saldo;
			
	}
		
	public Usuario (String Nombre, int Cedula, int Saldo, ArrayList<Boleta> Boletas) {
		this.Nombre = Nombre;
		this.Cedula = Cedula;
		this.Membresia = false;
		this.Saldo = Saldo;
		this.Boletas =Boletas;
			
	}
	//get con boolean	
		public boolean verificarMembresia() {
			
			return this.Membresia;
			
		}
		
		public void comprarMembresia() {
			
			
			this.Membresia = true;
			
		}
	//metodos get y set	
		public String getNombre() {
			
			return 	this.Nombre;
			
		}
		
		public void setNombre(String Nombre) {
			
			this.Nombre = Nombre;
			
		}

		public int getCedula() {
			
			return 	this.Cedula;
			
		}
		
		public void setCedula(int Cedula) {
			
			this.Cedula = Cedula;
			
		}

		public int getSaldo() {
			
			return 	this.Saldo;
			
		}

		public void setSaldo(int Saldo) {
			
			this.Saldo = Saldo;
			
		}

		public void agregarSaldo(int Saldo) {
			
			this.Saldo += Saldo;
			
		}

		public void agregarBoleta(Boleta Boleta) {
			
			this.Boletas.add(Boleta);
			
		}

		public ArrayList<Boleta> getBoletas() {
			
			return this.Boletas;
			
		}

		public void setBoletas(ArrayList<Boleta> Boletas) {
			
			this.Boletas = Boletas;
			
		}

		public HashMap<String, Integer> getCarrito() {
			
			return this.Carrito;
			
		}

		public void setCarrito(HashMap<String, Integer> Carrito) {
			
			this.Carrito = Carrito;
			
		}

		public void agregarCarrito(String Nombre, int Cantidad) {
			
			this.Carrito.put(Nombre, Cantidad);
			
		}

		public void eliminarCarrito(String Nombre) {
			
			this.Carrito.remove(Nombre);
			
		}

		public void vaciarCarrito() {
			
			this.Carrito.clear();
			
		}	

		//add a Boleta to the list of Boletas
		public void addBoleta(Boleta boleta) {
			
			this.Boletas.add(boleta);
			
		}	
		
}	

File: Python/src/gestionAplicacion/cinepy/Usuario.py
class Usuario:
    def __init__(self,nombre,cedula,membresia,saldo,boletas,carrito,cantidadUsuarios):
        self._nombre = nombre
        self._cedula = cedula
        self._membresia = membresia
        self._saldo = saldo
        self._boletas = boletas
        self._carrito = carrito
        self._cantidadUsuarios = cantidadUsuarios

    def getcantidadUsuarios(self):
            return self._cantidadUsuarios

    def __init__(self,nombre,cedula,saldo):
        self._nombre = nombre
        self._cedula = cedula
        self._saldo = saldo

    def __init__(self,nombre,cedula,saldo,boletas):
        self._nombre = nombre
        self._cedula = cedula
        self._saldo = saldo
        self._boletas = boletas
    
    def verificacionMemnresia(self):
        if self._membresia == True:
            return True
        else:
            return False

    def getNombre(self):
        return self._nombre
    
    def setNombre(self,nombre):
        self._nombre = nombre

    def getCedula(self):
        return self._cedula
    
    def setCeluda(self,cedula):
        self._cedula = cedula
    
    
    def getSaldo(self):
        return self._saldo
    
    def setSaldo(self,saldo):
        self._saldo = saldo
    
    def addSaldo(self,saldo):
        self._saldo += saldo

    def addBoleta(self,boleta):
        self._boletas.append(boleta)
    
    def getBoletas(self):
        return self._boletas
    
    def setBoletas(self,boletas):
        self._boletas = boletas
   
    def getcarrito(self):
        return self._carrito
            
    def setCarrito(self,carrito):
        self._carrito = carrito

    def addCarrito(self,carrito):
        self._carrito.append(carrito)

    def eliminarCarrito(self,carrito):
        self._carrito.remove(carrito)

    def vaciarCarrito(self):
        self._carrito = []

    def getBoletas(self):
        return self._boletas

    
    